### ISSUE TITLE
[BUG] HPO tutorial's `PyTorchLightningPruningCallback` fails due to `on_init_start` callback hook 

### ISSUE THREAD

**Issue by erinboyle** [unit8co/darts#1352]
**Describe the bug**
Executing the tutorial [here](https://unit8co.github.io/darts/examples/17-hyperparameter-optimization.html#Using-Optuna) fails with: 
```
The `on_init_start` callback hook was deprecated in v1.6 and is no longer supported as of v1.8.
``` 
Looks like `darts==0.22.0` should not have `pytorch-lightning>=1.5.0` but rather `pytorch-lightning==1.5.0`. (Default installation of darts currently grabs `pytorch-lightning==1.8.0.post1`.) So far rolling back lightning to 1.5.0 seems to be working.

**To Reproduce**
1. Default pip installation of darts 0.22.0
2. Run the HPO tutorial above

**Expected behavior**
Functional optuna x lightning pruning. In the shorter term, a fix to the HPO tutorial -- clarify either pinning to an earlier pytorch lightning (or maybe earlier darts), or replacing the pruning callback with something else if there's a good alternative. (Would be too bad to just eliminate it entirely.)

**System (please complete the following information):**
 - Python version: 3.8.14
 - darts version 0.22.0

**Comment by dennisbader** [comment]
Hi @erinboyle and thanks for reporting.

This is an issue with Optuna not supporting the latest PyTorch Lightning version.
It seems they are working on a fix [here](https://github.com/optuna/optuna/issues/3418).

Darts supports the latest version of PyTorch Lightning and we should keep it that way. I think the most simple way for now is to keep track of the fix, and maybe adding an info text to the HPO example, that at the time of writing Optuna only supports pytorch-lightning>=1.5.0.

**Comment by erinboyle** [comment]
@dennisbader - That makes sense. (Modulo typo - Optuna only supports <= 1.5.0.)

**Comment by madtoinou** [comment]
Hi @erinboyle,

Optuna's PR was merged in February and was released with version 3.2.0. I just ran the notebook with the latest darts version (0.25.0, with pytorch 2.0.5 and optuna 3.3.0) and it works just fine.

Thank you for reporting the bug with all these details! Feel free to reopen if the problem persists on your side.
